subprojects/language-scala/src/integTest/groovy/org/gradle/language/scala/CachedPlatformScalaCompileIntegrationTest.groovy

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.scala
+
+import org.gradle.api.tasks.compile.AbstractCachedCompileIntegrationTest
+import org.gradle.integtests.fixtures.RepoScriptBlockUtil
+import org.gradle.integtests.fixtures.TestResources
+import org.gradle.test.fixtures.file.TestFile
+import org.junit.Rule
+
+class CachedPlatformScalaCompileIntegrationTest extends AbstractCachedCompileIntegrationTest {
+    public static final String PLAY_REPOSITORIES = """
+        repositories {
+            ${RepoScriptBlockUtil.jcenterRepositoryDefinition()}
+            ${RepoScriptBlockUtil.lightbendMavenRepositoryDefinition()}
+            ${RepoScriptBlockUtil.lightbendIvyRepositoryDefinition()}
+        }
+    """
+
+    String compilationTask = ':compilePlayBinaryScala'
+    String compiledFile = "build/playBinary/classes/Hello.class"
+
+    @Rule
+    public final TestResources resources = new TestResources(temporaryFolder)
+
+    @Override
+    def setupProjectInDirectory(TestFile project = temporaryFolder.testDirectory) {
+        project.with {
+            file('settings.gradle') << localCacheConfiguration()
+            file('build.gradle').text = """
+            plugins {
+                id 'play'
+            }
+            ${PLAY_REPOSITORIES}
+            """.stripIndent()
+
+            file('app/controller/Hello.scala').text = """
+            object Hello {
+                def main(args: Array[String]): Unit = {
+                    print("Hello!")
+                }
+            }
+        """.stripIndent()
```

Comment:
This probably needs to be indented 4 spaces more. I would leave out the `.stripIndent()`, since it only makes the test a little bit harder to read.
I prefer indenting multiline strings like this:
```
    def multiLineString = """
        This is the text
        which goes over multiple lines
    """
```